pkgs/applications/misc/feedbackd/default.nix

```diff
@@ -61,7 +62,11 @@ stdenv.mkDerivation rec {
     libgudev
   ];
 
-  mesonFlags = [ "-Dgtk_doc=true" "-Dman=true" ];
+  mesonFlags = [
+    "-Dgtk_doc=${lib.boolToString (stdenv.buildPlatform == stdenv.hostPlatform)}"
+    "-Dman=true"
+    "-Dintrospection=${if (stdenv.buildPlatform == stdenv.hostPlatform) then "enabled" else "disabled"}"
```

Comment:
I am honestly not completely sure.

Reply:
I did a quick readup. We are basically missing the GObject bindings to other languages.
What does that mean in practical terms? For example you couldn't load them in python anymore. I am not sure what could break there but I imagine that maybe you cannot load the library there or parse all DBus messages.

So sounds to me like that programs in languages other than C that use Gnome related things will break and GUI programs written in Python could break in unexpected ways.

https://source.puri.sm/Librem5/feedbackd/-/blob/master/libfeedback/meson.build#L83-118

https://source.puri.sm/Librem5/feedbackd/-/blob/master/doc/overview.xml#L28-30

TLDR: We should fix this rather than adding this everywhere.

@jtojnar did I misunderstand something?

Reply:
Yeah, without introspection, the library cannot be used from languages like Python or JavaScript (well there is always manual FFI – but hopefully no project is masochistic enough to do that). It should fail with an error like the following at runtime so hopefully people will notice:

```Python traceback
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/nix/store/4lrl4lnlkigfb6jnv3p7qix337yz6zya-python3-3.9.12-env/lib/python3.9/site-packages/gi/importer.py", line 136, in load_module
    raise ImportError('cannot import name %s, '
ImportError: cannot import name Lfb, introspection typelib not found
```

Fixing it properly is hard, see #72868.

Reply:
Yeah, kinda depends on which application needs to use this and how they do it.